c10/core/TensorImpl.h

```diff
@@ -819,7 +835,7 @@ struct C10_API TensorImpl : public c10::intrusive_ptr_target {
    * True if a tensor allows changes to its metadata (e.g. sizes / strides / storage / storage_offset).
    */
   virtual bool allow_tensor_metadata_change() const {
-    return allow_tensor_metadata_change_;
+    return !opaque_handle_ && allow_tensor_metadata_change_;
```

Comment:
I don't think here needs to check opaque_handle_, since the only place that can generate opaque tensor is the new TensorImpl ctor, which has set `allow_tensor_metadata_change_` to `false` already.

Reply:
@bddppq `set_allow_tensor_metadata_change` would change the value and it is a public method. Do you prefer to add an assertion in it?

Reply:
@jgong5 Yes I would prefer to instead add an assertion to the setter.

Reply:
@bddppq OK - will change other similar code patterns accordingly too.

Reply:
@bddppq There are multiple places in `torch/csrc/autograd/variable.h` which call `set_allow_tensor_metadata_change(true)`. Now I just ignore the change inside `set_allow_tensor_metadata_change()` function but don't think it is a right approach. Do you mind I do the checking and changes from the caller sites in `torch/csrc/autograd/variable.h` instead?